src/modules/endpoint/DefaultDataContainer.js

```diff
@@ -106,8 +106,8 @@ class DefaultDataContainer implements DataContainer {
   }
 
   setCities = async (cities: Array<CityModel>) => {
-    this._cities = cities
-    return this._databaseConnector.storeCities(cities)
+    const cache = this.caches.cities
+    await cache.cache(cities, new DatabaseContext())
```

Comment:
Not sure whether I like a DatabaseContext with null values, but I think it is acceptable :)

Reply:
You mean with only null values? I mean you introduced language to be nullable, so as we have data that is independent of the city as well, it makes sense to have one with city being null as well :)